js/latoken.js

```diff
@@ -0,0 +1,715 @@
+        return result;
+    }
+
+    calculateFee (symbol, side, amount, price, takerOrMaker = 'taker') {
+        let market = this.markets[symbol];
+        let key = 'quote';
+        let rate = market[takerOrMaker];
+        let cost = amount * rate;
+        let precision = market['precision']['price'];
+        if (side === 'sell') {
+            cost *= price;
+        } else {
+            key = 'base';
+            precision = market['precision']['amount'];
+        }
+        cost = this.decimalToPrecision (cost, ROUND, precision, this.precisionMode);
+        return {
+            'type': takerOrMaker,
+            'currency': market[key],
+            'rate': rate,
+            'cost': parseFloat (cost),
+        };
+    }
+
+    async fetchBalance (params = undefined) {
+        await this.loadMarkets ();
+        if (params !== undefined) {
+            let balance = await this.privateGetAccountBalances(params);
+            let result = {};
+            result.currencyId = balance['currencyId'];
+            result.symbol = balance['symbol'];
+            result.name = balance['name'];
+            result.amount = balance['amount'];
+            result.available = balance['available'];
+            result.frozen = balance['frozen'];
+            result.pending = balance['pending'];
+            return {
+                'id': result.currencyId,
+                'symbol': result.symbol,
+                'name': result.name,
+                'amount': result.amount,
+                'available': result.available,
+                'frozen': result.frozen,
+                'pending': result.pending,
+            }
+        }
+        let balances = await this.privateGetAccountBalances(params);
+        let result = [];
+        for (let i = 0; i < balances.length; i++) {
+            let balance = balances[i];
+            let currencyId = balance['currencyId'];
+            let symbol = balance['symbol'];
+            let name = balance['name'];
+            let amount = this.safeFloat(balance, 'amount');
+            let available = this.safeFloat(balance, 'available');
+            let frozen = this.safeValue(balance, 'frozen'); 
+            let pending = this.safeValue(balance, 'pending');
+            result.push({
+                'currencyId': currencyId,
+                'symbol': symbol,
+                'name': name,
+                'amount': amount,
+                'available': available,
+                'frozen': frozen,
+                'pending': pending,
+            });
+        }
+        return result;
+    }
+
+    parseBidAsk (bidask) {
+        let price = parseFloat (bidask['price'])
+        let amount = parseFloat (bidask['amount'])
+        return { 
+            'price': price,
+            'amount': amount, 
+        }
+    }
+
+    parseBidsAsks (bidasks, priceKey = 0, amountKey = 1) {
+        return Object.values (bidasks || []).map (bidask => this.parseBidAsk (bidask, priceKey, amountKey))
```

Comment:
`.map` is not allowed. I think you can remove this method since it is already defined in the base class.

Reply:
I rewrote this method from exchange.js

Reply:
@PolozovT the base classes (Exchange.js, exchange.py, Exchange.php) are not transpiled, but the derived exchange classes are. Therefore the syntax differs and in the base classes it's more relaxed.

Reply:
done